File: Libreria/src/Complex.java
import java.util.*;

public class Complex {
	
	private double real;
	private double imag;
	
	public Complex() {
		real = 0.0;
		imag = 0.0;
	}
	
	public Complex(double real, double imag) {
		this.real = real;
		this.imag = imag;
	}
	
	
	public static double modulo(Complex c) {
		return Math.sqrt(Math.pow(c.real,2)+Math.pow(c.imag,2));
	}
	
	public static Complex conjugado(Complex c) {
		double x = -c.imag;
		return new Complex(c.real , x);
	}
	
	public double getImag() {
		return imag;
	}
	
	public double getReal() {
		return real;
	}
	
	public static double fase(Complex c) {
		double angulo = Math.atan((c.imag/c.real));
		return angulo;
	}
	
	public static void impresion(Complex c) {
		if(c.imag == 0) {
			System.out.println(c.real);
			return;
		}if(c.real == 0 ) {
			if (c.imag == 1) {
				System.out.println("i");
				return;
			}else if(c.imag == -1) {
				System.out.println( "-i");
				return;
			}
			System.out.println(c.imag+"i");
			return;
		}
		else if(c.imag > 0) {
			if (c.imag == 1) {
				System.out.println(c.real + " + i");
				return;
			}
			System.out.println(c.real + " + " + c.imag +"i");
			return;
		}
		else if(c.imag == -1) {
			System.out.println(c.real + " - i");
			return;
		}
		System.out.println(c.real+" - "+ Math.abs(c.imag) +"i");
		return;
	}
	
	
	
	public static void main(String[] args) {
		Complex c1 = new Complex(3.0,-2.0);
		Complex c2 = new Complex(1.0,2.0);
		
		Complex resultado = ComplexMath.producto(c1,c2);
		//System.out.println(resultado.real + " " + resultado.imag);
		impresion(resultado);
		Vector<Vector<Complex>> v = new Vector<Vector<Complex>>();
		v.addElement(c1);
		
	}
}

File: LibreriaPython/Complex.py
from math import *
from ComplexMath import *
from Simuladores import *

def modulo(c):
    return sqrt((c[0]**2)+(c[1]**2))

def conjugado(c):
    return (c[0],-c[1])

def fase(c):
    return atan((c[1]/c[0]))

def impresion(c):
    aux = c[1]
    if c[1] == 0:
        return str(c[0])
    if c[1] > 0:
        aux = " + " + str(c[1])
        if c[1] == 1:
            return str(c[0])+ " + i"
        return str(c[0])+ " + " +str(c[1]) +"i"
    elif c[1]==-1:
        return str(c[0])+ " - i"
    return str(c[0])+" - "+ str(abs(c[1])) +"i"
